### PR TITLE
NPE in JDTQuarkusManager.processConfigGroup

### LINKED ISSUE #60: NPE in JDTQuarkusManager.processConfigGroup
Using a project with this pom.xml:
```xml
<?xml version="1.0"?>
<project xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd" xmlns="http://maven.apache.org/POM/4.0.0"
    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
  <modelVersion>4.0.0</modelVersion>
  <groupId>foo.bar</groupId>
  <artifactId>toto</artifactId>
  <version>0.1.0-SNAPSHOT</version>
  <properties>
    <surefire-plugin.version>2.22.0</surefire-plugin.version>
    <maven.compiler.target>1.8</maven.compiler.target>
    <quarkus.version>0.21.1</quarkus.version>
    <maven.compiler.source>1.8</maven.compiler.source>
    <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
    <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
  </properties>
  <dependencyManagement>
    <dependencies>
      <dependency>
        <groupId>io.quarkus</groupId>
        <artifactId>quarkus-bom</artifactId>
        <version>${quarkus.version}</version>
        <type>pom</type>
        <scope>import</scope>
      </dependency>
    </dependencies>
  </dependencyManagement>
  <dependencies>
    <dependency>
      <groupId>org.spockframework</groupId>
      <artifactId>spock-core</artifactId>
      <version>1.3-groovy-2.5</version>
    </dependency>
    <dependency>
      <groupId>io.quarkus</groupId>
      <artifactId>quarkus-resteasy</artifactId>
    </dependency>
    <dependency>
      <groupId>io.quarkus</groupId>
      <artifactId>quarkus-junit5</artifactId>
      <scope>test</scope>
    </dependency>
    <dependency>
      <groupId>io.rest-assured</groupId>
      <artifactId>rest-assured</artifactId>
      <scope>test</scope>
    </dependency>
    <dependency>
      <groupId>io.quarkus</groupId>
      <artifactId>quarkus-hibernate-orm-panache</artifactId>
    </dependency>
    <dependency>
      <groupId>io.quarkus</groupId>
      <artifactId>quarkus-undertow</artifactId>
    </dependency>
    <dependency>
      <groupId>org.projectlombok</groupId>
      <artifactId>lombok</artifactId>
      <version>1.18.8</version>
    </dependency>
    <dependency>
      <groupId>io.quarkus</groupId>
      <artifactId>quarkus-keycloak</artifactId>
    </dependency>
  </dependencies>
  <build>
    <plugins>
      <plugin>
        <groupId>io.quarkus</groupId>
        <artifactId>quarkus-maven-plugin</artifactId>
        <version>${quarkus.version}</version>
        <executions>
          <execution>
            <goals>
              <goal>build</goal>
            </goals>
          </execution>
        </executions>
      </plugin>
      <plugin>
        <artifactId>maven-surefire-plugin</artifactId>
        <version>${surefire-plugin.version}</version>
        <configuration>
          <systemProperties>
            <java.util.logging.manager>org.jboss.logmanager.LogManager</java.util.logging.manager>
          </systemProperties>
        </configuration>
      </plugin>
    </plugins>
  </build>
  <profiles>
    <profile>
      <id>native</id>
      <activation>
        <property>
          <name>native</name>
        </property>
      </activation>
      <build>
        <plugins>
          <plugin>
            <groupId>io.quarkus</groupId>
            <artifactId>quarkus-maven-plugin</artifactId>
            <version>${quarkus.version}</version>
            <executions>
              <execution>
                <goals>
                  <goal>native-image</goal>
                </goals>
                <configuration>
                  <enableHttpUrlHandler>true</enableHttpUrlHandler>
                </configuration>
              </execution>
            </executions>
          </plugin>
          <plugin>
            <artifactId>maven-failsafe-plugin</artifactId>
            <version>${surefire-plugin.version}</version>
            <executions>
              <execution>
                <goals>
                  <goal>integration-test</goal>
                  <goal>verify</goal>
                </goals>
                <configuration>
                  <systemProperties>
                    <native.image.path>${project.build.directory}/${project.build.finalName}-runner</native.image.path>
                  </systemProperties>
                </configuration>
              </execution>
            </executions>
          </plugin>
        </plugins>
      </build>
    </profile>
  </profiles>
</project>
```
completion fails in application.properties with:
```
[Trace - 11:14:15 AM] Sending request 'textDocument/completion - (1)'.
Params: {
    "textDocument": {
        "uri": "file:///Users/fbricon/Dev/demos/here/toto/src/main/resources/application.properties"
    },
    "position": {
        "line": 2,
        "character": 1
    },
    "context": {
        "triggerKind": 1
    }
}


[Trace - 11:14:15 AM] Received request 'quarkus/projectInfo - (3)'.
Params: {
    "uri": "file:///Users/fbricon/Dev/demos/here/toto/src/main/resources/application.properties"
}


[Trace - 11:14:15 AM] Sending response 'quarkus/projectInfo - (3)'. Processing request took 1617ms
[Trace - 11:14:15 AM] Received response 'textDocument/completion - (1)' in 1627ms. Request failed: Request quarkus/projectInfo failed with message:  (-32603).
[Error - 11:14:15 AM] Request textDocument/completion failed.
  Message: Request quarkus/projectInfo failed with message: 
  Code: -32603 
```
If you look at the java server logs, you can see quarkus.jdt.core failed because of this NPE:
```
[Error - 11:14:15 AM] 29-Aug-2019 11:14:15 AM Error in calling delegate command handler
null
java.lang.NullPointerException
    at com.redhat.quarkus.jdt.core.JDTQuarkusManager.processConfigGroup(JDTQuarkusManager.java:326)
    at com.redhat.quarkus.jdt.core.JDTQuarkusManager.processMap(JDTQuarkusManager.java:547)
    at com.redhat.quarkus.jdt.core.JDTQuarkusManager.processConfigGroup(JDTQuarkusManager.java:415)
    at com.redhat.quarkus.jdt.core.JDTQuarkusManager.processConfigGroup(JDTQuarkusManager.java:363)
    at com.redhat.quarkus.jdt.core.JDTQuarkusManager.processConfigRoot(JDTQuarkusManager.java:208)
    at com.redhat.quarkus.jdt.core.JDTQuarkusManager.processQuarkusAnnotation(JDTQuarkusManager.java:157)
    at com.redhat.quarkus.jdt.core.JDTQuarkusManager.access$0(JDTQuarkusManager.java:148)
    at com.redhat.quarkus.jdt.core.JDTQuarkusManager$1.acceptSearchMatch(JDTQuarkusManager.java:126)
    at org.eclipse.jdt.internal.core.search.matching.MatchLocator.report(MatchLocator.java:2166)
    at org.eclipse.jdt.internal.core.search.matching.ClassFileMatchLocator.matchAnnotations(ClassFileMatchLocator.java:351)
    at org.eclipse.jdt.internal.core.search.matching.ClassFileMatchLocator.matchAnnotations(ClassFileMatchLocator.java:334)
    at org.eclipse.jdt.internal.core.search.matching.ClassFileMatchLocator.locateMatches(ClassFileMatchLocator.java:170)
    at org.eclipse.jdt.internal.core.search.matching.MatchLocator.process(MatchLocator.java:1965)
    at org.eclipse.jdt.internal.core.search.matching.MatchLocator.locateMatches(MatchLocator.java:1336)
    at org.eclipse.jdt.internal.core.search.matching.MatchLocator.locateMatches(MatchLocator.java:1373)
    at org.eclipse.jdt.internal.core.search.matching.MatchLocator.locateMatches(MatchLocator.java:1492)
    at org.eclipse.jdt.internal.core.search.JavaSearchParticipant.locateMatches(JavaSearchParticipant.java:112)
    at org.eclipse.jdt.internal.core.search.BasicSearchEngine.findMatches(BasicSearchEngine.java:250)
    at org.eclipse.jdt.internal.core.search.BasicSearchEngine.search(BasicSearchEngine.java:595)
    at org.eclipse.jdt.core.search.SearchEngine.search(SearchEngine.java:670)
    at com.redhat.quarkus.jdt.core.JDTQuarkusManager.getQuarkusProjectInfo(JDTQuarkusManager.java:118)
    at com.redhat.quarkus.jdt.internal.core.ls.QuarkusDelegateCommandHandler.getQuarkusProjectInfo(QuarkusDelegateCommandHandler.java:97)
    at com.redhat.quarkus.jdt.internal.core.ls.QuarkusDelegateCommandHandler.executeCommand(QuarkusDelegateCommandHandler.java:58)
    at org.eclipse.jdt.ls.core.internal.handlers.WorkspaceExecuteCommandHandler$1.run(WorkspaceExecuteCommandHandler.java:206)
    at org.eclipse.core.runtime.SafeRunner.run(SafeRunner.java:45)
    at org.eclipse.jdt.ls.core.internal.handlers.WorkspaceExecuteCommandHandler.executeCommand(WorkspaceExecuteCommandHandler.java:196)
    at org.eclipse.jdt.ls.core.internal.handlers.JDTLanguageServer.lambda$3(JDTLanguageServer.java:459)
    at org.eclipse.jdt.ls.core.internal.handlers.JDTLanguageServer.lambda$41(JDTLanguageServer.java:909)
    at java.util.concurrent.CompletableFuture.uniApply(CompletableFuture.java:602)
    at java.util.concurrent.CompletableFuture$UniApply.tryFire(CompletableFuture.java:577)
    at java.util.concurrent.CompletableFuture$Completion.exec(CompletableFuture.java:443)
    at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:289)
    at java.util.concurrent.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1056)
    at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1692)
    at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:157)

```
Null check is missing from:
https://github.com/redhat-developer/quarkus-ls/blob/25a5b079d9a0d27584f0e362a71eefa1f05929f7/quarkus.jdt/com.redhat.quarkus.jdt.core/src/main/java/com/redhat/quarkus/jdt/core/JDTQuarkusManager.java#L326

### PR DESCRIPTION
**Pull request by angelozerr**
Fixes #60

Signed-off-by: azerr <azerr@redhat.com>

### CODE REVIEW

File: quarkus.jdt/com.redhat.quarkus.jdt.core/src/main/java/com/redhat/quarkus/jdt/core/JDTQuarkusManager.java [hunk 1]
```diff
@@ -363,76 +363,76 @@ private static void processConfigGroup(IJavaElement javaElement, String baseKey,
...
+					location, source, enumerations, configPhase, quarkusProperties);
+		} else if ("double".equals(fieldTypeName)) {
+			addField(propertyName, type, ConfigItem.NO_DEFAULT.equals(defaultValue) ? "0" : defaultValue, docs,
+					location, source, enumerations, configPhase, quarkusProperties);
+		} else if ("float".equals(fieldTypeName)) {
+			addField(propertyName, type, ConfigItem.NO_DEFAULT.equals(defaultValue) ? "0" : defaultValue, docs,
+					location, source, enumerations, configPhase, quarkusProperties);
+		} else if (isMap(fieldTypeName)) {
+			// FIXME: find better mean to check field is a Map
+			// this code works only if user uses Map as declaration and not if it declare
```
Comment (fbricon) [inline quarkus.jdt/com.redhat.quarkus.jdt.core/src/main/java/com/redhat/quarkus/jdt/core/JDTQuarkusManager.java]: if they declare

File: quarkus.jdt/com.redhat.quarkus.jdt.core/src/main/java/com/redhat/quarkus/jdt/core/JDTQuarkusManager.java [hunk 2]
```diff
@@ -363,76 +363,76 @@ private static void processConfigGroup(IJavaElement javaElement, String baseKey,
...
+		} else if ("int".equals(fieldTypeName)) {
+			addField(propertyName, type, ConfigItem.NO_DEFAULT.equals(defaultValue) ? "0" : defaultValue, docs,
+					location, source, enumerations, configPhase, quarkusProperties);
+		} else if ("long".equals(fieldTypeName)) {
+			addField(propertyName, type, ConfigItem.NO_DEFAULT.equals(defaultValue) ? "0" : defaultValue, docs,
+					location, source, enumerations, configPhase, quarkusProperties);
+		} else if ("double".equals(fieldTypeName)) {
+			addField(propertyName, type, ConfigItem.NO_DEFAULT.equals(defaultValue) ? "0" : defaultValue, docs,
+					location, source, enumerations, configPhase, quarkusProperties);
+		} else if ("float".equals(fieldTypeName)) {
```
Comment (fbricon) [inline quarkus.jdt/com.redhat.quarkus.jdt.core/src/main/java/com/redhat/quarkus/jdt/core/JDTQuarkusManager.java]: can be 1 if condition grouping the 4 cases

Comment (fbricon) [reply quarkus.jdt/com.redhat.quarkus.jdt.core/src/main/java/com/redhat/quarkus/jdt/core/JDTQuarkusManager.java]:
```java
if (isNumber(fieldTypeName)) {...}`
```

isNumber can be basically implemented as Array.asList("short","int","long","double","float").contains(fieldTypeName)

Comment (angelozerr) [reply quarkus.jdt/com.redhat.quarkus.jdt.core/src/main/java/com/redhat/quarkus/jdt/core/JDTQuarkusManager.java]: @fbricon I agree with you, but I have done this choice (not group the 4 cases) to have similar code of the Quarkus ConfigDefinition https://github.com/quarkusio/quarkus/blob/master/core/deployment/src/main/java/io/quarkus/deployment/configuration/ConfigDefinition.java#L257

If you think we should have even 1 condition, I will update it.

Comment (fbricon) [reply quarkus.jdt/com.redhat.quarkus.jdt.core/src/main/java/com/redhat/quarkus/jdt/core/JDTQuarkusManager.java]: "short" type is missing still

Comment (angelozerr) [reply quarkus.jdt/com.redhat.quarkus.jdt.core/src/main/java/com/redhat/quarkus/jdt/core/JDTQuarkusManager.java]: That's strange teh Quarkus ConfigDefinition doesn't manage this case?

Comment (fbricon) [reply quarkus.jdt/com.redhat.quarkus.jdt.core/src/main/java/com/redhat/quarkus/jdt/core/JDTQuarkusManager.java]: you're right, I didn't check

Comment (angelozerr) [reply quarkus.jdt/com.redhat.quarkus.jdt.core/src/main/java/com/redhat/quarkus/jdt/core/JDTQuarkusManager.java]: @fbricon do I must change something?

Comment (fbricon) [reply quarkus.jdt/com.redhat.quarkus.jdt.core/src/main/java/com/redhat/quarkus/jdt/core/JDTQuarkusManager.java]: I'd still prefer 1 condition as it's less prone to errors when maintaining the code

Comment (angelozerr) [reply quarkus.jdt/com.redhat.quarkus.jdt.core/src/main/java/com/redhat/quarkus/jdt/core/JDTQuarkusManager.java]: @fbricon fixed

File: quarkus.jdt/com.redhat.quarkus.jdt.test/src/main/java/com/redhat/quarkus/jdt/core/JDTQuarkusManagerDeploymentTest.java [hunk 3]
```diff
@@ -45,10 +46,60 @@
  */
 public class JDTQuarkusManagerDeploymentTest {
 
+	static {
```
Comment (fbricon) [inline quarkus.jdt/com.redhat.quarkus.jdt.test/src/main/java/com/redhat/quarkus/jdt/core/JDTQuarkusManagerDeploymentTest.java]: prefer setting that in @BeforeClass and restoring the original level in @AfterClass, so other tests are not impacted

File: quarkus.jdt/com.redhat.quarkus.jdt.test/src/main/java/com/redhat/quarkus/jdt/core/JDTQuarkusManagerTest.java [hunk 4]
```diff
@@ -31,6 +34,10 @@
 
 	private static final String QUARKUS_CORE_DEPLOYMENT_JAR = getJarPath("quarkus-core-deployment-0.19.1.jar.test");
 
+	static {
```
Comment (fbricon) [inline quarkus.jdt/com.redhat.quarkus.jdt.test/src/main/java/com/redhat/quarkus/jdt/core/JDTQuarkusManagerTest.java]: same thing. If you need it for multiple classes, then better to use a base test class that fiddles with log levels

File: quarkus.jdt/com.redhat.quarkus.jdt.test/src/main/java/com/redhat/quarkus/jdt/core/JDTQuarkusManagerDeploymentTest.java (hunk 3 above)
Comment (angelozerr) [reply quarkus.jdt/com.redhat.quarkus.jdt.test/src/main/java/com/redhat/quarkus/jdt/core/JDTQuarkusManagerDeploymentTest.java]: Done with new class BaseJDTQuarkusManagerTest

File: quarkus.jdt/com.redhat.quarkus.jdt.test/src/main/java/com/redhat/quarkus/jdt/core/JDTQuarkusManagerTest.java (hunk 4 above)
Comment (angelozerr) [reply quarkus.jdt/com.redhat.quarkus.jdt.test/src/main/java/com/redhat/quarkus/jdt/core/JDTQuarkusManagerTest.java]: Done with new class BaseJDTQuarkusManagerTest

**Pull request merged**